src/platform/patmos/s4noc_channel.c

```diff
@@ -0,0 +1,136 @@
+#include "reactor-uc/logging.h"
+#include "reactor-uc/serialization.h"
+#include <machine/patmos.h>
+#include <stdint.h>
+
+#define S4NOC_CHANNEL_ERR(fmt, ...) LF_ERR(NET, "S4NOCPollChannel: " fmt, ##__VA_ARGS__)
+#define S4NOC_CHANNEL_WARN(fmt, ...) LF_WARN(NET, "S4NOCPollChannel: " fmt, ##__VA_ARGS__)
+#define S4NOC_CHANNEL_INFO(fmt, ...) LF_INFO(NET, "S4NOCPollChannel: " fmt, ##__VA_ARGS__)
+#define S4NOC_CHANNEL_DEBUG(fmt, ...) LF_DEBUG(NET, "S4NOCPollChannel: " fmt, ##__VA_ARGS__)
+
+S4NOCGlobalState s4noc_global_state;
+
+static lf_ret_t S4NOCPollChannel_open_connection(NetworkChannel *untyped_self) {
+  S4NOC_CHANNEL_DEBUG("Open connection");
+  (void)untyped_self;
+  return LF_OK;
+}
+
+static void S4NOCPollChannel_close_connection(NetworkChannel *untyped_self) {
+  S4NOC_CHANNEL_DEBUG("Close connection");
+  (void)untyped_self;
+}
+
+static void S4NOCPollChannel_free(NetworkChannel *untyped_self) {
+  S4NOC_CHANNEL_DEBUG("Free");
+  (void)untyped_self;
+}
+
+static bool S4NOCPollChannel_is_connected(NetworkChannel *untyped_self) {
+  S4NOCPollChannel *self = (S4NOCPollChannel *)untyped_self;
+
+  return self->state == NETWORK_CHANNEL_STATE_CONNECTED;
+}
+
+static lf_ret_t S4NOCPollChannel_send_blocking(NetworkChannel *untyped_self, const FederateMessage *message) {
+  S4NOCPollChannel *self = (S4NOCPollChannel *)untyped_self;
+
+  volatile _IODEV int *s4noc_data = (volatile _IODEV int *)(PATMOS_IO_S4NOC + 4);
+  volatile _IODEV int *s4noc_dest = (volatile _IODEV int *)(PATMOS_IO_S4NOC + 8);
+
+  if (self->state == NETWORK_CHANNEL_STATE_CONNECTED) {
+    int message_size = serialize_to_protobuf(message, self->write_buffer + 4, S4NOC_CHANNEL_BUFFERSIZE - 4);
+    
+    *((int *)self->write_buffer) = message_size;
+
+    int total_size = message_size + 4;
+    *s4noc_dest = self->destination_core;
```

Comment:
uhh I dont know if this plays nicely if you have multiple S4NOC Channels sending to different cores.